packer/build.go

```diff
@@ -341,15 +341,15 @@ PostProcessorRunSeqLoop:
 			// Exception: for postprocessors that will fail/become
 			// useless if keep isn't true, heed forceOverride and keep the
 			// input artifact regardless of user preference.
-			if corePP.keepInputArtifact != nil {
-				if defaultKeep && *corePP.keepInputArtifact == false && forceOverride {
+			if corePP.KeepInputArtifact != nil {
+				if defaultKeep && *corePP.KeepInputArtifact == false && forceOverride {
```

Comment:
I was looking for statement where we actually set `*corePP.KeepInputArtifact` to `true`. But it looks like that is being handled via the line 336 `keep := defaultKeep` is that correct?  Does `*corePP.KeepInputArtifact` need to be set to ensure we get the right value somewhere later on?

Reply:
`*corePP.KeepInputArtifact` is set in the configuration steps parsing the HCL2 or JSON config.

defaultKeep is `*true`, but then if the `KeepInputArtifact` (which is a pointer) is not nil then that means the user explicitly set it and we set `keep` with that value.